src/lib.rs

```diff
@@ -91,13 +95,13 @@ impl Agent {
         }
     }
 
-    async fn send_metric(&self, val: MetricValue) {
-        let metric = HostMetricWrapper(&self.host_id, val).into();
-        // TODO: error handling.
-        let result = self.client.post_metrics(metric).await;
-        if result.is_err() {
-            dbg!(result.err());
-        }
+    async fn send_metric(&mut self, val: MetricValue) {
+        let mut metric: Vec<_> = HostMetricWrapper(&self.host_id, val).into();
+        metric.extend(self.metric.clone());
```

Comment:
It should limit the size of metrics, otherwise it stops after reaching the maximum body size.

Reply:
For reference, metrics sender of container agent keeps the metrics for 6 hours and sends metrics in 3 minutes at most.
https://github.com/mackerelio/mackerel-container-agent/blob/36cf0bb103d1645fb733c214fe142207b1318838/metric/sender.go#L22-L50

Reply:
Exactly. After reaching, should Agent drop a metric from the most previous for heaping new one?

Reply:
Thank you for your comment! If you are still interested in, please review #23 too.